contrib/src/main/java/com/datatorrent/contrib/parser/StreamingJsonParser.java

```diff
@@ -0,0 +1,451 @@
+import com.datatorrent.lib.parser.Parser;
+import com.datatorrent.lib.util.FieldInfo;
+import com.datatorrent.lib.util.FieldInfo.SupportType;
+import com.datatorrent.lib.util.KeyValPair;
+import com.datatorrent.lib.util.PojoUtils;
+
+/**
+ * Operator that parses a JSON string tuple and emits a POJO on the output port
+ * and tuples that could not be parsed on error port.Upstream operator needs to
+ * ensure that a full JSON record is emitted.<br>
+ * <b>Properties</b><br>
+ * <b>pojoClass</b>:POJO class <br>
+ * <b>(optional)fieldMappingString</b>String of format
+ * fieldNameInJson:fieldNameInPOJO:DataType<br>
+ * <b>Ports</b> <br>
+ * <b>in</b>:input tuple as a String. Each tuple represents a json string<br>
+ * <b>out</b>:tuples that are validated as per the user defined POJO are emitted
+ * as POJO on this port<br>
+ * <b>err</b>:tuples that could not be parsed are emitted on this port as
+ * KeyValPair<String,String><br>
+ * Key being the tuple and Val being the reason
+ * 
+ * @displayName SimpleStreamingJsonParser
+ * @category Parsers
+ * @tags json pojo parser streaming
+ */
+@InterfaceStability.Evolving
+public class StreamingJsonParser extends Parser<byte[], KeyValPair<String, String>>
+{
+  private String jsonSchema;
+  private transient JSONParser jsonParser;
+  private String fieldMappingString;
+  private List<FieldInfo> fieldInfos;
+  private List<ActiveFieldInfo> columnFieldSetters;
+  protected JsonKeyFinder finder;
+  private static final String FIELD_SEPARATOR = ":";
+  private static final String RECORD_SEPARATOR = ",";
+  private ArrayList<String> columnFields;
+  private transient Class<?> pojoClass;
+
+  /**
+   * @return POJO class
+   */
+  private Class<?> getPojoClass()
+  {
+    return pojoClass;
+  }
+
+  /**
+   * Sets the POJO class
+   */
+  public void setPojoClass(Class<?> pojoClass)
+  {
+    this.pojoClass = pojoClass;
+  }
+
+  /**
+   * Returns a string representing mapping from generic record to POJO fields
+   */
+  public String getFieldMappingString()
+  {
+    return fieldMappingString;
+  }
+
+  /**
+   * Comma separated list mapping a field in JSON schema to POJO field eg :
+   * fieldName:INTEGER,fieldName:DOUBLE
+   */
+  public void setFieldMappingString(String genericRecordToPOJOFieldsMapping)
+  {
+    this.fieldMappingString = genericRecordToPOJOFieldsMapping;
+  }
+
+  public String getJsonSchema()
+  {
+    return jsonSchema;
+  }
+
+  public void setJsonSchema(String jsonSchema)
+  {
+    this.jsonSchema = jsonSchema;
+  }
+
+  public StreamingJsonParser()
+  {
+
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    jsonParser = new JSONParser();
+    finder = new JsonKeyFinder();
+    columnFields = new ArrayList<String>();
+    columnFieldSetters = Lists.newArrayList();
+
+    //use reflection if POJO to JSON key mapping is missing
+    if (getFieldMappingString() == null) {
+      setFieldInfos(createFieldInfoMap(generateFieldInfoInputs(getPojoClass())));
```

Comment:
Need to set the match key list on the JsonKeyFinder in setup.
POJO class is required for this  since the pojoToJsonMapping is optional.This is also why setPojo is present and public.